File: DevelopStudyDemo/app/src/main/java/com/chang/android/develop/MainActivity.java
package com.chang.android.develop;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.chang.android.develop.locationaware.LocationAwareActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void onLocationAwareClick(View view) {
        startActivity(new Intent(this, LocationAwareActivity.class));
    }
}

File: AIDLDemo/remoteserver/src/main/java/com/chang/android/aidl/server/MainActivity.kt
package com.chang.android.aidl.server

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View

class MainActivity : AppCompatActivity() {

    companion object {
        private const val TAG = "MainActivity"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun onOpenRemoteServer(view: View) {
        Log.d(TAG, "点击开启远程服务")
        startService(RemoteService.getIntent(this))
    }

    override fun onDestroy() {
        super.onDestroy()
        stopService(RemoteService.getIntent(this))
    }
}